src/main/java/uk/ac/cam/cl/dtg/segue/api/AdminFacade.java

```diff
@@ -1250,25 +1250,21 @@ public Response getDiagnostics(@Context final Request request, @Context final Ht
      * @return Confirmation of success, or error message on incorrect role.
      */
     @POST
-    @Path("/reset_misuse_monitor/{event_label}")
+    @Path("/reset_misuse_monitor/{event_label}/{agent_identifier}")
     @Produces(MediaType.APPLICATION_JSON)
     @Operation(summary = "Reset a misuse monitor counter to zero.")
     public Response resetMisuseMonitor(@Context final HttpServletRequest request,
                                        @PathParam("event_label") final String eventLabel,
-                                       final String agentIdentifier) {
+                                       @PathParam("agent_identifier") final String agentIdentifier) {
         try {
             RegisteredUserDTO user = userManager.getCurrentRegisteredUser(request);
             if (!isUserAnAdmin(userManager, user)) {
                 return SegueErrorResponse.getIncorrectRoleResponse();
             }
-            if (misuseMonitor.hasMisused(agentIdentifier, eventLabel)) {
-                misuseMonitor.resetMisuseCount(agentIdentifier, eventLabel);
-                log.info(String.format("Admin user (%s) reset misuse monitor '%s' for user (%s)!", user.getEmail(),
-                        eventLabel, agentIdentifier));
-                return Response.ok(ImmutableMap.of("status", "Reset successfully!")).build();
-            } else {
-                return Response.ok(ImmutableMap.of("status", "Nothing to reset.")).build();
```

Comment:
This was removed because I removed the `misuseMonitor.hasMisused(...)` condition, since it makes sense for admins to be able to reset the misuse count at any point, not just after is becomes misused. Because of this, the only response would be "Reset successfully!" which IMO is just as good as a 200.

Reply:
But I need to know I have actually reset the thing I was intending to reset? If I can't find it in the new UI, I need to know that I have got the right (encoding of the) agent identifier and without this there's no way an admin can tell any longer!

Reply:
I see that it's nice to be able to reset it _before_ it crosses the threshold, though. Maybe it's fine like this. I certainly don't want any more time spent on this.